### PR TITLE
Make sure the stream backend playback/library handle URIs the same way

### PR DESCRIPTION
**Pull request by adamcik**
Probably replaces #1446. Main thing missing for 2.0 is a changelog, but it's time for bed now...

### CODE REVIEW

File: tests/stream/test_library.py [hunk 1]
```diff
@@ -30,26 +36,28 @@ def track_uri():
     return path.path_to_uri(path_to_data_dir('song1.wav'))
 
 
-def test_lookup_ignores_unknown_scheme(backend):
-    library = actor.StreamLibraryProvider(backend, [])
+def test_lookup_ignores_unknown_scheme(audio, config):
+    backend = actor.StreamBackend(audio=audio, config=config)
+    backend.library.lookup('http://example.com') == []
```
Comment (jodal) [inline tests/stream/test_library.py]: Missing `assert`

File: mopidy/stream/actor.py [hunk 2]
```diff
@@ -43,27 +52,25 @@ def __init__(self, config, audio):
...
             return []
 
-        if self._blacklist_re.match(uri):
+        if self.backend._blacklist_re.match(uri):
             logger.debug('URI matched metadata lookup blacklist: %s', uri)
             return [Track(uri=uri)]
 
-        try:
-            result = self._scanner.scan(uri)
+        result = _unwrap_stream(
```
Comment (jodal) [inline mopidy/stream/actor.py]: Wouldn't it be nicer with `_, scan_result = ...` ?

File: mopidy/stream/actor.py [hunk 3]
```diff
@@ -71,23 +78,22 @@ def lookup(self, uri):
...
-
     def translate_uri(self, uri):
+        if urllib.parse.urlsplit(uri).scheme not in self.backend.uri_schemes:
+            return None
+
+        if self.backend._blacklist_re.match(uri):
+            logger.debug('URI matched metadata lookup blacklist: %s', uri)
+            return uri
+
         return _unwrap_stream(
```
Comment (jodal) [inline mopidy/stream/actor.py]: Wouldn't it be nicer with `uri, _ = ...` and then `return uri` ?

**Comment by jodal** [comment]
:+1: with comments and a changelog entry

**Comment by adamcik** [comment]
Should get to it this evening.

**Pull request merged**